lib/src/rules/pub/no_http_urls.dart

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/src/lint/pub.dart'; // ignore: implementation_imports
+
+import '../../analyzer.dart';
+
+const _desc = r"Don't use http urls.";
+
+const _details = r'''
+**DON'T** use http urls in `pubspec.yaml`.
+
+Use https instead.
+
+**GOOD:**
+```yaml
+repository: 'https://github.com/dart-lang/example'
+```
+
+**BAD:**
+```dart
+repository: 'http://github.com/dart-lang/example'
+''';
+
+class PubspecNoHttpUrls extends LintRule {
+  PubspecNoHttpUrls()
+      : super(
+            name: 'pubspec_no_http_urls',
+            description: _desc,
+            details: _details,
+            group: Group.pub);
+
+  @override
+  PubspecVisitor getPubspecVisitor() => Visitor(this);
+
+  @override
+  LintCode get lintCode => const LintCode("Don't use http urls.",
```

Comment:
The first argument needs to be the name of the lint ('pubspec_no_http_urls').

Reply:
Done